File: typescript/__tests__/problems101_200/binary-tree-right-side-view.test.ts
import {deepEqual} from 'assert';
import TreeNode from '../../src/concepts/tree-node';
import SOLUTIONS from '../../src/problems101_200/binary-tree-right-side-view';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    root: number[],
  },
  output: number[],
  message: string,
}

describe(
    'BinaryTreeRightSideView',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        deepEqual(
                            solution.rightSideView(TreeNode.of(...sample.input.root)),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);

File: java/src/test/java/problems101_200/BinaryTreeRightSideViewTest.java
package problems101_200;

import concepts.TreeNode;
import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class BinaryTreeRightSideViewTest extends SampledTest {
    @Test
    public void test() {
        for (BinaryTreeRightSideView solution : BinaryTreeRightSideView.values()) {
            for (BinaryTreeRightSideViewSample sample
                : getSamples(BinaryTreeRightSideViewSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.rightSideView(TreeNode.of(sample.input.root)))
                    .containsExactlyElementsIn(Arrays.stream(sample.output).boxed().toList())
                    .inOrder();
            }
        }
    }
}
